xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -180,7 +182,8 @@ private DownstreamTlsContext getDownstreamTlsContext(InetSocketAddress localInet
       FilterChainComparator comparator = new FilterChainComparator(localInetAddr);
       FilterChain bestMatch =
           filterChains.isEmpty() ? null : Collections.max(filterChains, comparator);
-      if (bestMatch != null && comparator.isMatching(bestMatch.getFilterChainMatch())) {
+      if (bestMatch != null
+          && (newServerApi || comparator.isMatching(bestMatch.getFilterChainMatch()))) {
```

Comment:
Not related too this PR, but some comment on the code:

This is really an abused usage of `Comparator`. All you need is just a static helper method returning true if two filter chains are _equivalent_. `Comparator` is for implementing objects _total ordering_, the overridden `compare` method makes zero sense for ordering a collection of `FilterChain` objects.

Reply:
Let me explain what this is doing: it is not for comparing 2 filter chains. It is for ordering the filters in a filter chain based on the attributes of the channel such as IP address and port (both local and remote). So the `Comparator` is used to rank the filters in the filter chain to find the best match for the given channel e.g. An exact IP address match is better than a match with IP_any (0.0.0.0). Admittedly we don't need a complex `Comparator` to achieve this and a simple scoring based algorithm would have worked but the reviewer wanted it. 

Since the Comparator just ranks the filters in the chain we still need to make sure the top ranked filter fits the current channel which is where the `comparator.isMatching()` is made on the `bestMatch`.